portfolio/src/main/java/com/google/sps/servlets/VideoServlet.java

```diff
@@ -19,21 +19,16 @@
     private static int QUERY_SIZE = 7;
 
     @Override
-    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{
-        String videoId = request.getParameter(PropertyNames.VIDEO_ID);
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException{
 
-        // Have the user enter video metadata until we can fetch it.
-        String title = request.getParameter(PropertyNames.TITLE);
-        String description = request.getParameter(PropertyNames.DESCRIPTION);
-        String captions = request.getParameter(PropertyNames.CAPTIONS);
-        Video dummyVideo = new Video(videoId, title, description, captions);
-        
+        // Create the video and load its captions.
+        Video video = Video.httpRequestToVideo(request);
+        video.loadCaptions();
```

Comment:
Can we put this in the httpRequestToVideo function?

Reply:
We discussed this a bit in our integration meeting today. The main reason behind loading captions separately is that they are not needed for when we preview a Video for users (to confirm that they want to analyze it) and captions are actually a lot of data. However, we weren't completely sure about this decision and are definitely open to other ideas.

Reply:
I see. Loading captions is an API call which is expensive. Acknowledged, SGTM.